base/channels.jl

```diff
@@ -454,6 +455,156 @@ function put_unbuffered(c::Channel, v)
     return v
 end
 
+"""
+    append!(c::Channel, iter)
+
+Append all items in `iter` to the channel `c`. If the channel is buffered, this operation requires
+fewer `lock` operations than individual `put!`s. Blocks if the channel is full.
```

Comment:
Would be nice to mention: If there is room for 3 more elements and you try to append 4, will it block before adding any elements, or after adding the first 3?

Reply:
I'm not sure I understand the question. the `append!` call will not return until all elements in `iter` are added to the channel. Internally, it releases the lock on the channel when the buffer is full so that `take!` calls can run to make more room.

I expanded this sentence a bit. Can you see if it adds the clarity you're looking for?